File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        differenceHashMap = {}
        for index, num in enumerate(nums):
            difference = target - num
            if difference in differenceHashMap:
                return [differenceHashMap[difference], index]
            differenceHashMap[num] = index

File: Data Structures & Algorithms/majority-element/submission-2.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority_element = 0, count = 0;

        for (int num: nums) {
            if (count == 0) {
                majority_element = num;
            }
            if (majority_element == num) {
                count++;
            } else {
                count--;
            }
        }
        return majority_element;
    }
};